### DISCUSSION TITLE
How to Use Early Stopping in test/evaluate

### DISCUSSION THREAD

**Discussion by yanbing-j** [Lightning-AI/pytorch-lightning#14135] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi pytorch-lightning team,

I want to early stop after several iterations in inference in trainer.test(). However, in training, `on_train_batch_start()` can return -1 to early stop. In test, using `on_test_batch_start` and return -1 cannot help. Does pytorch-lightning has this function in inference? Could you please share some test cases? Thank you!

**Comment by rohitgr7** [comment]
EarlyStopping isn't supposed to be used during test/evaluation. Its sole purpose is to stop the training during convergence.

**Reply by yanbing-j** [reply]
Thank you for your clarification!

In performance profiling, EarlyStopping is equal important in test/evaluation. Will you take this into account?

**Reply by rohitgr7** [reply]
can you explain more on `performance profiling + early stopping` ?

**Reply by yanbing-j** [reply]
Sure. For example, I want to optimize the inference part of the model, that equals to test/evaluation part. Now I can only run one complete epoch to profile, which is to large to store the trace down. Therefore, I want to only record the profiling in several iterations with enough warmup. EarlyStopping can help doing this.

**Reply by rohitgr7** [reply]
you can limit the batches there using `limit_val_batches/limit_test_batches`.
